src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -117,9 +118,9 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
          * This improves the recall.
          */
         if (filterWeight != null && canDoExactSearch(filterIdsArray.length)) {
-            docIdsToScoreMap.putAll(doExactSearch(context, filterIdsArray));
+            docIdsToScoreMap.putAll(doExactSearch(context, filterIdsArray, knnQuery.getParentsFilter()));
         } else {
-            Map<Integer, Float> annResults = doANNSearch(context, filterIdsArray);
+            Map<Integer, Float> annResults = doANNSearch(context, filterIdsArray, knnQuery.getParentsFilter());
```

Comment:
you don't need to pass the parent filters to these functions. As parent filter is part of kNNQueryObject you do `this.knnQuery.getParentFilter()` at any place where you need to get the parent filter.

Reply:
ack